stdlib/src/sys/ffi.mojo

```diff
@@ -383,39 +419,93 @@ fn _get_dylib_function[
 # ===----------------------------------------------------------------------===#
 
 
-@always_inline("nodebug")
+@always_inline
 fn external_call[
     callee: StringLiteral, type: AnyTrivialRegType, *types: AnyType
-](*arguments: *types) -> type:
+](*args: *types) -> type:
     """Calls an external function.
 
     Args:
-      arguments: The arguments to pass to the external function.
+        args: The arguments to pass to the external function.
 
     Parameters:
-      callee: The name of the external function.
-      type: The return type.
-      types: The argument types.
+        callee: The name of the external function.
+        type: The return type.
+        types: The argument types.
 
     Returns:
-      The external call result.
+        The external call result.
     """
+    return external_call[callee, type](args)
+
 
-    # The argument pack will contain references for each value in the pack,
-    # but we want to pass their values directly into the C printf call. Load
-    # all the members of the pack.
-    var loaded_pack = _LITRefPackHelper(arguments._value).get_loaded_kgen_pack()
+@always_inline("nodebug")
+fn external_call[
+    callee: StringLiteral, type: AnyTrivialRegType
+](args: VariadicPack) -> type:
+    """Calls an external function.
+
+    Args:
+        args: The arguments to pass to the external function.
+
+    Parameters:
+        callee: The name of the external function.
+        type: The return type.
+
+    Returns:
+        The external call result.
+    """
+    var p = args._get_loaded_kgen_pack()
 
     @parameter
     if _mlirtype_is_eq[type, NoneType]():
-        __mlir_op.`pop.external_call`[func = callee.value, _type=None](
-            loaded_pack
-        )
+        __mlir_op.`pop.external_call`[func = callee.value, _type=None](p)
         return rebind[type](None)
     else:
-        return __mlir_op.`pop.external_call`[func = callee.value, _type=type](
-            loaded_pack
-        )
+        return __mlir_op.`pop.external_call`[func = callee.value, _type=type](p)
+
+
+# FIXME: no one should be having to do manual pop.external_call for extra_args
```

Comment:
I don't think we need to include this now (or in the future when unpacking syntax becomes available).